File: momose/src/models/randomWalkMessageModel/RandomWalkMessageNode.java
package models.randomWalkMessageModel;

import java.util.ArrayList;

import engine.SimTime;
import math.Point2d;
import models.messageModel.MessageNode;


public class RandomWalkMessageNode extends MessageNode
{
    protected float localTime;
    protected float vMax;
    protected float vMin;
    protected float pauseTime;
    
    public RandomWalkMessageNode(Point2d pos, float radius, int id, SimTime time) 
    {
        super(pos, radius, id, time);
        
        localTime = 0;
        vMax=5;
        vMin=0;
        pauseTime=1f;
    } // end constructor
    
    public void think(SimTime time) {
        randomMove(time);
    }
    
    private void randomMove(SimTime time)
    {
        if((localTime>=pauseTime)||(collided==true)) 
        { 
            // Calculate the new velocity vector
            v.x=(float)(Math.random()*vMax+vMin)*(Math.random()<0.5?(-1):(1));
            v.y=(float)(Math.random()*vMax+vMin)*(Math.random()<0.5?(-1):(1));
            
            // Calculate the new velocity
            v.x=v.x*time.getDT();
            v.y=v.y*time.getDT();
            
            localTime=0;  
        } 
        
        // increment the local time 
        localTime+=time.getDT();
        
    } // end randomMove
} // end MessageNode

File: momosecpp/src/RandomWalkMessageNode.cpp
#include <stdlib.h>
#include <stdio.h>
#include <time.h>
#include "RandomWalkMessageNode.h"

RandomWalkMessageNode::RandomWalkMessageNode(Point2D pos, const float &radius, int nodeId, SimTime *simTime): MessageNode(pos, radius, nodeId, simTime)
{
  localTime = 0;
 
  //  srand((unsigned) time(NULL));
}

RandomWalkMessageNode::~RandomWalkMessageNode(){}

void RandomWalkMessageNode::think(SimTime *simTime) 
{
  randomMove(simTime);	
}

void RandomWalkMessageNode::randomMove(SimTime *simTime)
{
  static int count = 0;

  if((localTime >= pauseTime) || (collided == true)) 
    {	
      v.x = (float)((float)rand()/(float)(RAND_MAX)) * (vMax + vMin)
   	                * (rand() % 2 < 1 ? (-1) : (1));
      v.y = (float)((float)rand()/(float)(RAND_MAX)) * (vMax + vMin)
   	                * (rand() %2 < 1 ? (-1) : (1));
      v*=simTime->getDT();
	 
      localTime=0;  
    }	
  
  localTime+=simTime->getDT();    
 }
